aptos-move/framework/aptos-framework/sources/delegation_pool.move

```diff
@@ -888,6 +896,119 @@ module aptos_framework::delegation_pool {
         }
     }
 
+    /// Initialize a delegation pool from an existing owned staking contract.
+    /// The newly created delegation pool will preserve the operator and commission fee. The staker becomes the pool
+    /// owner and has the same delegated voter set.
+    public entry fun initialize_delegation_pool_from_staking_contract(
+        staker: &signer,
+        operator: address,
+    ) acquires DelegationPool, GovernanceRecords, BeneficiaryForOperator, NextCommissionPercentage {
+        assert!(features::delegation_pools_enabled(), error::invalid_state(EDELEGATION_POOLS_DISABLED));
+        assert!(
+            features::staking_contract_to_delegation_pool_conversion_enabled(),
+            error::invalid_state(ESTAKING_CONTRACT_TO_DELEGATION_POOL_CONVERSION_NOT_SUPPORTED)
+        );
+
+        let staker_address = signer::address_of(staker);
+        assert!(!owner_cap_exists(staker_address), error::already_exists(EOWNER_CAP_ALREADY_EXISTS));
+
+        // do not have to withhold the `add_stake` fee from staker if there is no pending_active stake
```

Comment:
Can you add some documentation before the function definition to list the conditions for when a staking contract can be turned into a delegation pool? This includes this condition here of not having any pending active stake